src/ert/run_models/base_run_model.py

```diff
@@ -847,6 +853,64 @@ def _evaluate_and_postprocess(
         return num_successful_realizations
 
 
+class HasDesignParameters:
```

Comment:
This class makes the entire design matrix very explicit. This PR
https://github.com/equinor/ert/pull/11071

will make design matrix parameters only a attribute of parameter; ie. it does not need to be explicit anymore.